pkg/network/ephemeral_darwin.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
Why do we need this file? NPM is not supported on mac.

Reply:
For running tests locally on mac developer machines.  

I'm open to suggestions here.  I considered using the `test` build flag but I was concerned that it would conflict with the `linux` build flag when the tests run in CI.

Reply:
We usually confine testing and dev to the agent VM, so this is not really required.

Reply:
when you run tests locally on your dev machine you'll run them in the linux VM anyway, so I don't think this will be used

Reply:
For drive-by/infrequent contributors, setting up the agent VM and having to run tests there introduces a lot of friction when for a lot of these tests it is not required.

Would it be more acceptable to scope this to the `test` build tag instead?

Reply:
The agent tests for NPM require a linux environment since most of them use ebpf. I don't think we should add code to support running tests on mac; it also will confuse developers about whether we support mac or not.

Reply:
>I don't think we should add code to support running tests on mac;

There is a mac os build step that is [now failing](https://github.com/DataDog/datadog-agent-macos-build/actions/runs/6739757703/job/18321800503) upon removing this code because it is now falling back to `ephemeral_unsupported` which always returns false for `IsPortInEphemeralRange`

What do you prefer to move forward here?

Reply:
Add `//go:build linux || windows` to `state_test.go`